pwndbg/dbg/__init__.py

```diff
@@ -251,6 +254,39 @@ def remove(self) -> None:
         raise NotImplementedError()
 
 
+class EventType(Enum):
+    """
+    Events that can be listened for and reacted to in a debugger.
+
+    The events types listed here are defined as follows:
+        - `START`: This event is fired some time between the creation of or
+          attachment to the process to be debugged, and the start of its
```

Comment:
Are we 100%  sure it is also called on attach?

Reply:
On GDB, it's called the first time `new_objfile` fires and there's a selected thread. The GDB tests expect [newly attached-to processes to have their symbol tables read](https://sourceware.org/git/?p=binutils-gdb.git;a=blob;f=gdb/testsuite/gdb.base/attach.exp;h=2b655e69f8badf31b7f66ec49f881737ccd7bb60;hb=HEAD), which implies that at least the objfile for the main executable will always be loaded. And experimentally I can confirm that it does seem to always get loaded. Meaning that we get a `start` event, in the same way that we would if were starting a process.

On LLDB, we fire that event ourselves, so we can guarantee that's the case.